src/test/javascript/spec/component/lecture/pdf-preview.component.spec.ts

```diff
@@ -0,0 +1,255 @@
+        fixture.detectChanges();
+
+        expect(alertServiceSpy).toHaveBeenCalledOnce();
+    });
+
+    it('should handle errors and trigger alert when loading an attachment unit file fails', () => {
+        routeMock.data = of({
+            course: { id: 1, name: 'Example Course' },
+            attachmentUnit: { id: 1, name: 'Chapter 1' },
+        });
+        const errorResponse = new HttpErrorResponse({
+            status: 404,
+            statusText: 'Not Found',
+            error: 'File not found',
+        });
+
+        const attachmentUnitService = TestBed.inject(AttachmentUnitService);
+        jest.spyOn(attachmentUnitService, 'getAttachmentFile').mockReturnValue(throwError(() => errorResponse));
+        const alertServiceSpy = jest.spyOn(alertServiceMock, 'error');
+
+        component.ngOnInit();
+        fixture.detectChanges();
+
+        expect(alertServiceSpy).toHaveBeenCalledOnce();
+    });
+
+    it('should load PDF and verify rendering of pages', () => {
+        const mockBlob = new Blob(['PDF content'], { type: 'application/pdf' });
+
+        attachmentServiceMock.getAttachmentFile.mockReturnValue(of(mockBlob));
+        component.ngOnInit();
+
+        expect(URL.createObjectURL).toHaveBeenCalledWith(mockBlob);
+        expect(attachmentServiceMock.getAttachmentFile).toHaveBeenCalledWith(1, 1);
+        expect(component.totalPages).toBeGreaterThan(0);
+    });
+
+    it('should navigate through pages using keyboard in enlarged view', () => {
+        component.isEnlargedView = true;
+        component.totalPages = 5;
+        component.currentPage = 3;
+
+        const eventRight = new KeyboardEvent('keydown', { key: 'ArrowRight' });
+        const eventLeft = new KeyboardEvent('keydown', { key: 'ArrowLeft' });
+
+        component.handleKeyboardEvents(eventRight);
+        expect(component.currentPage).toBe(4);
+
+        component.handleKeyboardEvents(eventLeft);
+        expect(component.currentPage).toBe(3);
+    });
+
+    it('should toggle enlarged view state', () => {
+        const mockCanvas = document.createElement('canvas');
+        component.displayEnlargedCanvas(mockCanvas, 1);
+        expect(component.isEnlargedView).toBeTrue();
+
+        const clickEvent = new MouseEvent('click', {
+            button: 0,
+        });
+
+        component.closeEnlargedView(clickEvent);
+        expect(component.isEnlargedView).toBeFalse();
+    });
+
+    it('should prevent scrolling when enlarged view is active', () => {
+        component.toggleBodyScroll(true);
+        expect(component.pdfContainer.nativeElement.style.overflow).toBe('hidden');
+
+        component.toggleBodyScroll(false);
+        expect(component.pdfContainer.nativeElement.style.overflow).toBe('auto');
+    });
+
+    it('should not update canvas size if not in enlarged view', () => {
+        component.isEnlargedView = false;
+        component.currentPage = 3;
+
+        const spy = jest.spyOn(component, 'updateEnlargedCanvas');
```

Comment:
The client test build fails here because `updateEnlargedCanvas` is a private method

Reply:
Made public